DAO.sol

```diff
@@ -121,6 +128,13 @@ contract DAOInterface {
         bool newCurator;
         // Data needed for splitting the DAO
         SplitData[] splitData;
+        // Dependencies in other proposals. List of proposal IDs and list defining
+        // whether the other proposal needs to have passed or failed
+        uint[] dependencies;
```

Comment:
Why not do:

```js
struct Dependency {
    uint proposalID;
    bool shouldPasss;
}
```

and then have it as:

```js
Depency[] dependencies
```

inside the proposal struct?

If you insist on keeping it as is then change:
1. `dependencies` -> `dependentProposalIDs`
2. `dependenciesType` -> `dependencyShouldPass`

Because as it currently stands the name are confusing.